From f51e57044937c44e0c198973c90a429937a7fa53 Mon Sep 17 00:00:00 2001
From: Dinh Nguyen <dinguyen@altera.com>
Date: Wed, 26 Feb 2014 10:36:41 -0600
Subject: [PATCH 194/248] FogBugz #186724: Add backwards compatible binding for
 dw_apb_timer

In order to have the 3.10-ltsi kernel use the same DTS as the 3.13
kernel, we need to add the old compatibility string for the dw_apb_timer.
The 3.10 kernel was using the ""snps,dw-apb-timer-sp" and
"snps,dw-apb-timer-osc" compatible strings for the timers.

[Original patch taken from
git://git.rocketboards.org/linux-socfpga.git socfpga-3.14]

Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
Signed-off-by: Dinh Nguyen <dinguyen@altera.com>
---
 arch/arm/boot/dts/socfpga.dtsi | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/boot/dts/socfpga.dtsi b/arch/arm/boot/dts/socfpga.dtsi
index 04e96f0..f372532 100644
--- a/arch/arm/boot/dts/socfpga.dtsi
+++ b/arch/arm/boot/dts/socfpga.dtsi
@@ -794,7 +794,7 @@
 		};
 
 		timer0: timer0@ffc08000 {
-			compatible = "snps,dw-apb-timer";
+			compatible = "snps,dw-apb-timer", "snps,dw-apb-timer-sp";
 			interrupts = <0 167 4>;
 			reg = <0xffc08000 0x1000>;
 			clocks = <&l4_sp_clk>;
@@ -802,7 +802,7 @@
 		};
 
 		timer1: timer1@ffc09000 {
-			compatible = "snps,dw-apb-timer";
+			compatible = "snps,dw-apb-timer", "snps,dw-apb-timer-sp";
 			interrupts = <0 168 4>;
 			reg = <0xffc09000 0x1000>;
 			clocks = <&l4_sp_clk>;
@@ -810,7 +810,7 @@
 		};
 
 		timer2: timer2@ffd00000 {
-			compatible = "snps,dw-apb-timer";
+			compatible = "snps,dw-apb-timer", "snps,dw-apb-timer-osc";
 			interrupts = <0 169 4>;
 			reg = <0xffd00000 0x1000>;
 			clocks = <&osc1>;
@@ -818,7 +818,7 @@
 		};
 
 		timer3: timer3@ffd01000 {
-			compatible = "snps,dw-apb-timer";
+			compatible = "snps,dw-apb-timer", "snps,dw-apb-timer-osc";
 			interrupts = <0 170 4>;
 			reg = <0xffd01000 0x1000>;
 			clocks = <&osc1>;
-- 
1.9.1

